### ISSUE TITLE
A backend for QuantumCumulants.jl [$200]

### ISSUE THREAD

**Issue by Krastanov** [QuantumSavory/QuantumSymbolics.jl#73]
# A backend for QuantumCumulants.jl [$200]

The QuantumCumulants library (part of the QuantumOptics organization) provides for an elegant way to solve truncated differential equations for expectation values of quantum observables. To claim this bounty you need to create an `express` interface for converting symbolic objects from QuantumSymbolics into objects QuantumCumulants can work with.

**Required skills**: Knowledge of the QuantumCumulants library

**Reviewer**: Stefan Krastanov and/or @david-pl

**Duration**: 2 months

#### Payout procedure:

The Funding for these bounties comes from the National Science Foundation and from the NSF Center for Quantum Networks. The payouts are managed by the NumFOCUS foundation and processed in bulk once every two months. If you live in a country in which NumFOCUS can make payments, you can participate in this bounty program.

[Click here for more details about the bug bounty program.](https://github.com/QuantumSavory/.github/blob/main/BUG_BOUNTIES.md)

<details>
<summary><strong>Bug bounty logistic details</strong> (click to expand)</summary>

To claim exclusive time to work on this bounty either post a comment here or message [skrastanov@umass.edu](mailto:skrastanov@umass.edu) with:

- your name
- github username
- **(optional)** a brief list of previous pertinent projects you have engaged in

Currently the project is claimed by `no one` until `...`.

If you want to, you can work on this project without making a claim, however claims are encouraged to give you and other contributors peace of mind. Whoever has made a claim takes precedence when solutions are considered.

You can always propose your own funded project, if you would like to contribute something of value that is not yet covered by an official bounty.
</details>

**Comment by jdiaz97** [comment]
I request assignment to work on this bounty

**Comment by Krastanov** [comment]
Hi there! Thanks for your interest! Could you elaborate a bit on your experience with quantum information science or quantum mechanics? This will be a bit difficult to tackle without that background.

Let's also discuss what capabilities should be included in the scope of the bounty.

**Comment by jdiaz97** [comment]
Hi again! @Krastanov 
I don't have experience with quantum stuff. I mostly want to take this to learn about it actually. I think it's a good opportunity and I think I could get it done.

What specific capabilities do you think should be included in the bounty?

**Comment by Krastanov** [comment]
Happy to assign it if you are still interested. A good starting point would be:

- being able to create QuantumCumulants Hamiltonians from QuantumSavory
- being able to convert QuantumCumulants expressions to other backends
- being able to convert from other backends to QuantumCumulants

A word of warning though, this is probably a bit too niche of a project to serve as a good introduction to the field.

**Comment by limitless-100** [comment]
Hello @Krastanov , I would love to take this project this week.

**Comment by Krastanov** [comment]
I have not heard from @jdiaz97 about whether he is interested. Given what we discussed over email, @limitless-100 , please feel free proceed with this bounty. We need to make sure that this is actually useful, not just a pro-forma exercise, so do not hesitate to post here if it is unclear how to structure things or what capabilities are needed. I am marking this as reserved.

**Comment by limitless-100** [comment]
Is working on those three tasks you listed 16 days ago enough?

**Comment by limitless-100** [comment]
Is working on those three tasks you listed 16 days ago enough?

**Comment by Krastanov** [comment]
Yes, but it will take some effort to define them fully and have them tested, so if something is unclear just let me know (and expect some back and forth during the review stage of the proposal)

**Comment by Krastanov** [comment]
Yes, the goal is to use the express interface to be able to convert symbolic objects from QuantumSymbolics.jl to objects in QuantumCummulants.jl. All features of QuantumCummulants.jl should be supported, which might need making improvements to QuantumSymbolics.jl, but there are plenty of things in QuantumSymbolics.jl that do not have a counterpart in QuantumCummulants.jl -- none of them need to be touched.

**Comment by oameye** [comment]
Notice that these days QuantumCumalants.jl has moved its interface to SecondQuantizedAlgebra.jl.

**Comment by oameye** [comment]
We are working on a new ground up design of SecondQuantizedAlgebra.jl in https://github.com/qojulia/SecondQuantizedAlgebra.jl/pull/99. This should be merged soon but will break the internal representation. After that QC.jl will be upgraded
